scheduler/src/cook/mesos/api.clj

```diff
@@ -822,23 +828,34 @@
 (defn retrieve-jobs
   [conn ctx]
   (let [jobs (get-in ctx [:request :query-params :job])
-        instances (get-in ctx [:request :query-params :instance])]
+        instances (get-in ctx [:request :query-params :instance])
+        allow-partial-results (get-in ctx [:request :query-params :partial])]
     (let [instance-uuid->job-uuid #(instance-uuid->job-uuid (d/db conn) %)
           instance-jobs (mapv instance-uuid->job-uuid instances)
           used? (partial job-exists? (db conn))]
       (cond
-        (and (every? used? jobs)
-             (every? (complement nil?) instance-jobs))
+        (and (not allow-partial-results)
+             (every? used? jobs)
+             (every? some? instance-jobs))
         [true {::jobs (into jobs instance-jobs)
                ::jobs-requested jobs
                ::instances-requested instances}]
+
+        (and allow-partial-results
+             (or (some used? jobs)
+                 (some some? instance-jobs)))
+        [true {::jobs (into (filter used? jobs) (filter some? instance-jobs))
```

Comment:
`(remove nil? instance-jobs)` is cleared to me than `(filter some? instance-jobs)`